x/concentrated-liquidity/incentives.go

```diff
@@ -0,0 +1,83 @@
+package concentrated_liquidity
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/osmoutils/sumtree"
+	"github.com/osmosis-labs/osmosis/v13/x/concentrated-liquidity/model"
+	"github.com/osmosis-labs/osmosis/v13/x/concentrated-liquidity/types"
+)
+
+// Gets the current join-time sumtree for the passed in pool ID.
+func (k Keeper) getSumtree(ctx sdk.Context, poolID uint64) sumtree.Tree {
+	return sumtree.NewTree(prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyAccumulationStore(poolID)), 10)
+}
+
+// Adds to a position's liquidity in Liquidity tree, setting its JoinTime to current block time.
+//
+// Since we want to reset a position's JoinTime every time they update their liquidity, any
+// liquidity update is equivalent to removing the old JoinTime node from the liquidity tree
+// and replacing it with a new one. Thus, this function simply takes in the old and new liquidity
+// amounts and leaves liquidity delta calculations to the caller.
```

Comment:
Side note: this will require the user to claim any outstanding rewards prior to replacing the node from the liquidity tree right?